File: FPMS_Automation_Skeleton_Pack/playwright_ts/src/support/router.ts
import { test } from "@playwright/test";

import { annotateCase } from "./annotations";
import type { BoundaryCase, CaseHandler, ExecutionContext, TestCase } from "./types";
import { isSkeleton } from "./helpers";
import { waveW0Handlers } from "../handlers/waveW0";
import { waveAHandlers } from "../handlers/waveA";
import { waveBHandlers } from "../handlers/waveB";
import { waveCHandlers } from "../handlers/waveC";
import { waveG0Handlers } from "../handlers/waveG0";
import { waveDHandlers } from "../handlers/waveD";
import { waveEHandlers } from "../handlers/waveE";
import { waveFHandlers } from "../handlers/waveF";
import { waveGHandlers } from "../handlers/waveG";
import { waveHHandlers } from "../handlers/waveH";
import { waveXHandlers } from "../handlers/waveX";
import { boundaryHandlers } from "../handlers/boundary";

const handlers: Record<string, CaseHandler<any>> = {
  ...waveW0Handlers,
  ...waveAHandlers,
  ...waveBHandlers,
  ...waveCHandlers,
  ...waveG0Handlers,
  ...waveDHandlers,
  ...waveEHandlers,
  ...waveFHandlers,
  ...waveGHandlers,
  ...waveHHandlers,
  ...waveXHandlers,
  ...boundaryHandlers,
};

export async function executeCase(tc: TestCase, ctx: ExecutionContext): Promise<void> {
  annotateCase(tc);
  const handler = handlers[tc.id];
  test.skip(isSkeleton(handler), `Skeleton only: ${tc.id} | ${tc.topic}`);
  await handler(ctx, tc);
}

export async function executeBoundaryCase(tc: BoundaryCase, ctx: ExecutionContext): Promise<void> {
  annotateCase(tc);
  const handler = handlers[tc.id];
  test.skip(isSkeleton(handler), `Boundary skeleton only: ${tc.id} | ${tc.object}`);
  await handler(ctx, tc);
}

File: FPMS_Automation_Skeleton_Pack/pytest_python/framework/router.py
from __future__ import annotations

import pytest

from framework.runtime import RuntimeContext
from framework.models import BoundaryCase, TestCase
from handlers.wave_w0 import HANDLERS as w0_handlers
from handlers.wave_a import HANDLERS as a_handlers
from handlers.wave_b import HANDLERS as b_handlers
from handlers.wave_c import HANDLERS as c_handlers
from handlers.wave_g0 import HANDLERS as g0_handlers
from handlers.wave_d import HANDLERS as d_handlers
from handlers.wave_e import HANDLERS as e_handlers
from handlers.wave_f import HANDLERS as f_handlers
from handlers.wave_g import HANDLERS as g_handlers
from handlers.wave_h import HANDLERS as h_handlers
from handlers.wave_x import HANDLERS as x_handlers
from handlers.boundary import HANDLERS as boundary_handlers

ALL_HANDLERS = {
    **w0_handlers,
    **a_handlers,
    **b_handlers,
    **c_handlers,
    **g0_handlers,
    **d_handlers,
    **e_handlers,
    **f_handlers,
    **g_handlers,
    **h_handlers,
    **x_handlers,
    **boundary_handlers,
}


def execute_case(case: TestCase, runtime: RuntimeContext) -> None:
    handler = ALL_HANDLERS.get(case.id)
    if handler is None or getattr(handler, "_is_skeleton", False):
        pytest.skip(f"Skeleton only: {case.id} | {case.topic}")
    handler(runtime, case)


def execute_boundary_case(case: BoundaryCase, runtime: RuntimeContext) -> None:
    handler = ALL_HANDLERS.get(case.id)
    if handler is None or getattr(handler, "_is_skeleton", False):
        pytest.skip(f"Boundary skeleton only: {case.id} | {case.object}")
    handler(runtime, case)
